L1Trigger/CSCTriggerPrimitives/src/CSCCathodeLCTProcessor.cc

```diff
@@ -1256,13 +1250,51 @@ void CSCCathodeLCTProcessor::encodeHighMultiplicityBits(
   std::vector<unsigned> station_thresholds = {
       thresholds_[csc_idx * 3], thresholds_[csc_idx * 3 + 1], thresholds_[csc_idx * 3 + 2]};
 
-  // assign the bits
-  for (unsigned i = 0; i < station_thresholds.size(); i++) {
-    if (hitsInTime >= station_thresholds[i]) {
-      inTimeHMT_ = i + 1;
+  //hard coded dead time as 2Bx, since showerNumTBins = 3, like firmware
+  // for example, nhits = 0 at bx7; = 100 at bx8; = 0 at bx9
+  //cathode HMT must be triggered at bx8, not bx7 and bx9
+  //meanwhile we forced 2BX dead time after active shower trigger
+  unsigned int deadtime =
+      showerNumTBins_ - 1;  // firmware hard coded dead time as 2Bx, since showerNumTBins = 3 in firmware
+  unsigned int dead_count = 0;
+  bool dead_status = false;
+
+  for (unsigned bx = 0; bx < CSCConstants::MAX_CLCT_TBINS; bx++) {
+    unsigned minbx = bx >= showerNumTBins_ / 2 ? bx - showerNumTBins_ / 2 : bx;
+    unsigned maxbx = bx < CSCConstants::MAX_CLCT_TBINS - showerNumTBins_ / 2 ? bx + showerNumTBins_ / 2
+                                                                             : CSCConstants::MAX_CLCT_TBINS - 1;
+    unsigned this_hitsInTime = 0;
+    bool isPeak = true;  //check whether total hits in bx is peak of nhits over time bins
+    /*following is to count number of hits over [minbx, maxbx], showerNumTBins=3 =>[n-1, n+1]*/
+    for (unsigned mbx = minbx; mbx <= maxbx; mbx++) {
+      this_hitsInTime += hitsInTime[mbx];
     }
-  }
 
-  // create a new object
-  shower_ = CSCShowerDigi(inTimeHMT_, false, theTrigChamber);
+    if (peakCheck_ and bx < CSCConstants::MAX_CLCT_TBINS - showerNumTBins_ / 2 - 1) {
+      if (hitsInTime[minbx] < hitsInTime[maxbx + 1] or
+          (hitsInTime[minbx] == hitsInTime[maxbx + 1] and hitsInTime[bx] < hitsInTime[bx + 1]))
+        isPeak = false;  //next bx would have more hits or in the center
+    }
+    if (dead_count > 0) {
+      dead_count--;
+      dead_status = true;
+    } else
+      dead_status = false;
```

Comment:
`dead_status` could be a local variable in the scope:
```suggestion
    bool dead_status = dead_count > 0:
    if (dead_status)
      dead_count--;
```

Reply:
Fixed as suggested